src/quo2/components/selectors/selectors.cljs

```diff
@@ -1,109 +1,73 @@
 (ns quo2.components.selectors.selectors
   (:require [quo2.components.icon :as icons]
-            [quo2.foundations.colors :as colors]
+            [quo2.components.selectors.styles :as style]
             [react-native.core :as rn]
-            [reagent.core :as reagent]
-            [quo2.components.selectors.styles :as style]))
+            [reagent.core :as reagent]))
 
 (defn- handle-press
-  [disabled? on-change checked?]
-  (when (not disabled?)
-    (fn []
-      (swap! checked? not)
-      (when on-change (on-change @checked?)))))
+  [on-change checked-atom checked?]
+  (when checked-atom (swap! checked-atom not))
+  (when on-change (on-change (not checked?))))
 
-(defn checkbox-prefill
-  [{:keys [default-checked?]}]
-  (let [internal-checked? (reagent/atom (or default-checked? false))]
-    (fn [{:keys [on-change disabled? blurred-background? container-style checked?]}]
-      (when (and (not (nil? checked?)) (not= @internal-checked? checked?))
```

Comment:
Removed this atom update form all the selectors, the reason is I don't perceive an atom update directly in the body's component render function as a good practice. 

Why?
Well, if you update an atom in the body, that will trigger a re-render, so that the atom could be updated again, taking us to an infinite cycle. Finally to avoid this cycle we need to make sure to add a condition to not to update the atom (such as this `when` in this component), and usually, these conditions are hard to understand/debug.

IMO, it's a better practice to put the update inside callback-fns (on-change, on-press, etc).